jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/HumanTaskHandlerHelper.java

```diff
@@ -201,6 +216,72 @@ public static Deadlines setDeadlines(Map<String, Object> parameters, List<Organi
 		return notifications;
 	}
 
+	protected static int getDeadlineRepeatLimit(String deadlineStr) {
+		String[] repeatableParts = DateTimeUtils.parseISORepeatable(deadlineStr);
+		String repeatLimit = repeatableParts[0];
+		if (!repeatLimit.isEmpty()) {
+			return Integer.parseInt(repeatLimit);
+		} else {
+			return -1;
+		}
+	}
+
+	protected static Date getDeadlineDurationDate(String durationStr, int repeatCount) {
+		// handles iso repetable and both period and date notation
+		try {
+
+			if (DateTimeUtils.isRepeatable(durationStr)) {
+				String[] repeatableParts = DateTimeUtils.parseISORepeatable(durationStr);
+				String tempTimeDelay = repeatableParts[1];
+				String tempDateTimeStr = repeatableParts[2];
+
+				Date retDate;
+
+				if (DateTimeUtils.isPeriod((tempDateTimeStr))) {
+					// e.g R/start/duration
+					retDate = new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + repeatDuration(tempDateTimeStr, repeatCount));
+				} else {
+					if(DateTimeUtils.isPeriod(tempTimeDelay)) {
+						// e.g R/duration/end
+						long firstRet = DateTimeUtils.parseDateTime(tempDateTimeStr) -  Duration.parse(tempTimeDelay).toMillis();
+						retDate =  new Date(firstRet + repeatDuration(tempTimeDelay, repeatCount));
+					} else {
+						// e.g R/start/end
+						// duration is end - start
+						long duration = DateTimeUtils.parseDateTime(tempDateTimeStr) - DateTimeUtils.parseDateTime(tempTimeDelay);
+						retDate =  new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + repeatDuration(duration, repeatCount));
+					}
```

Comment:
This can be simplified to:

```
if (DateTimeUtils.isPeriod((tempDateTimeStr))) {
...
} else if(DateTimeUtils.isPeriod(tempTimeDelay)) {
...
} else {
...
}
```